### ISSUE TITLE
Uncaught Error: This TextEditor has been destroyed

### ISSUE THREAD

**Issue by egor-kamenev** [AtomLinter/linter-jscs#153]
[Enter steps to reproduce below:]

1. ...
2. ...

**Atom Version**: 1.3.3
**System**: Mac OS X 10.10.5
**Thrown From**: [linter-jscs](https://github.com/AtomLinter/linter-jscs) package, v3.2.2

### Stack Trace

Uncaught Error: This TextEditor has been destroyed

```
At /Users/egorkamenev/Downloads/Atom.app/Contents/Resources/app.asar/src/pane.js:829

Error: This TextEditor has been destroyed
    at DisplayBuffer.module.exports.DisplayBuffer.screenPositionForBufferPosition (/Users/egorkamenev/Downloads/Atom.app/Contents/Resources/app.asar/src/display-buffer.js:790:15)
    at TextEditorMarker.module.exports.TextEditorMarker.getHeadScreenPosition (/Users/egorkamenev/Downloads/Atom.app/Contents/Resources/app.asar/src/text-editor-marker.js:179:33)
    at Cursor.module.exports.Cursor.getScreenPosition (/Users/egorkamenev/Downloads/Atom.app/Contents/Resources/app.asar/src/cursor.js:71:26)
    at TextEditor.module.exports.TextEditor.getCursorScreenPosition (/Users/egorkamenev/Downloads/Atom.app/Contents/Resources/app.asar/src/text-editor.js:1432:35)
    at Function.fixString (/Users/egorkamenev/.atom/packages/linter-jscs/index.js:154:35)
    at /Users/egorkamenev/.atom/packages/linter-jscs/index.js:79:16
    at Function.module.exports.Emitter.simpleDispatch (/Users/egorkamenev/Downloads/Atom.app/Contents/Resources/app.asar/node_modules/event-kit/lib/emitter.js:25:14)
    at Emitter.module.exports.Emitter.emit (/Users/egorkamenev/Downloads/Atom.app/Contents/Resources/app.asar/node_modules/event-kit/lib/emitter.js:125:28)
    at TextBuffer.module.exports.TextBuffer.saveAs (/Users/egorkamenev/Downloads/Atom.app/Contents/Resources/app.asar/node_modules/text-buffer/lib/text-buffer.js:1060:27)
    at TextBuffer.module.exports.TextBuffer.save (/Users/egorkamenev/Downloads/Atom.app/Contents/Resources/app.asar/node_modules/text-buffer/lib/text-buffer.js:1030:19)
    at TextEditor.module.exports.TextEditor.save (/Users/egorkamenev/Downloads/Atom.app/Contents/Resources/app.asar/src/text-editor.js:602:26)
    at Pane.module.exports.Pane.saveItem (/Users/egorkamenev/Downloads/Atom.app/Contents/Resources/app.asar/src/pane.js:534:18)
    at Pane.module.exports.Pane.saveActiveItem (/Users/egorkamenev/Downloads/Atom.app/Contents/Resources/app.asar/src/pane.js:517:19)
    at Workspace.module.exports.Workspace.saveActivePaneItem (/Users/egorkamenev/Downloads/Atom.app/Contents/Resources/app.asar/src/workspace.js:633:35)
    at atom-workspace.commandRegistry.add.core:save (/Users/egorkamenev/Downloads/Atom.app/Contents/Resources/app.asar/src/register-default-commands.js:173:32)
    at CommandRegistry.module.exports.CommandRegistry.handleCommandEvent (/Users/egorkamenev/Downloads/Atom.app/Contents/Resources/app.asar/src/command-registry.js:260:29)
    at /Users/egorkamenev/Downloads/Atom.app/Contents/Resources/app.asar/src/command-registry.js:3:61
    at KeymapManager.module.exports.KeymapManager.dispatchCommandEvent (/Users/egorkamenev/Downloads/Atom.app/Contents/Resources/app.asar/node_modules/atom-keymap/lib/keymap-manager.js:536:16)
    at KeymapManager.module.exports.KeymapManager.handleKeyboardEvent (/Users/egorkamenev/Downloads/Atom.app/Contents/Resources/app.asar/node_modules/atom-keymap/lib/keymap-manager.js:359:22)
    at WindowEventHandler.module.exports.WindowEventHandler.handleDocumentKeydown (/Users/egorkamenev/Downloads/Atom.app/Contents/Resources/app.asar/src/window-event-handler.js:97:36)
    at HTMLDocument.<anonymous> (/Users/egorkamenev/Downloads/Atom.app/Contents/Resources/app.asar/src/window-event-handler.js:3:61)
```

### Commands

```
     -1:13.5.0 editor:move-to-end-of-screen-line (atom-text-editor.editor.is-focused)
     -1:12.8.0 core:move-up (atom-text-editor.editor.is-focused)
  2x -1:11.4.0 core:move-left (atom-text-editor.editor.is-focused)
  8x -1:10.7.0 core:select-right (atom-text-editor.editor.is-focused)
  3x -1:10 core:select-left (atom-text-editor.editor.is-focused)
     -0:40.3.0 core:save (atom-text-editor.editor.is-focused)
     -0:24.3.0 core:copy (atom-text-editor.editor.is-focused)
     -0:23.1.0 core:paste (atom-text-editor.editor.is-focused)
     -0:22.6.0 core:save (atom-text-editor.editor.is-focused)
     -0:14.3.0 editor:newline (atom-text-editor.editor.is-focused)
     -0:14 core:save (atom-text-editor.editor.is-focused)
     -0:12.5.0 core:backspace (atom-text-editor.editor.is-focused)
     -0:11.9.0 core:save (atom-text-editor.editor.is-focused)
     -0:02.6.0 editor:newline (atom-text-editor.editor.is-focused)
     -0:02.1.0 core:save (atom-text-editor.editor.is-focused)
     -0:00.4.0 core:backspace (atom-text-editor.editor.is-focused)
```

### Config

```json
{
  "core": {
    "themes": [
      "seti-ui",
      "monokai"
    ],
    "openEmptyEditorOnStart": false,
    "destroyEmptyPanes": false,
    "disabledPackages": [
      "linter-jshint"
    ]
  },
  "linter-jscs": {
    "displayAs": "jscs Error",
    "esnext": true,
    "fixOnSave": true,
    "onlyConfig": true,
    "configPath": ".jscs.json"
  }
}
```

### Installed Packages

```coffee
# User
linter, v1.11.3
linter-jscs, v3.2.2
monokai, v0.18.0
seti-ui, v0.8.1
terminal-plus, v0.14.5

# Dev
No dev packages
```

**Comment by Arcanemagus** [comment]
Do you have Fix on Save enabled?

**Comment by axit-rasmus** [comment]
With Fix on save enabled, I get the same thing sometimes when I save. Reloading the tab fixes it.

**Comment by casesandberg** [comment]
Can you update `atom` and `linter-jscs` and see if this is still an issue?

**Comment by axit-rasmus** [comment]
Can not consistently reproduce. It had been working fine all day (6+ hours) after updating, but apparently it's still here.

**Atom Version**: 1.4.0
**System**: Microsoft Windows 8.1 Pro
**Thrown From**: [linter-jscs](https://github.com/AtomLinter/linter-jscs) package, v3.4.0


### Stack Trace

Uncaught Error: This TextEditor has been destroyed

```
At C:\Users\Rasmus\AppData\Local\atom\app-1.4.0\resources\app.asar\src\pane.js:829

Error: This TextEditor has been destroyed
    at DisplayBuffer.module.exports.DisplayBuffer.screenPositionForBufferPosition (C:\Users\Rasmus\AppData\Local\atom\app-1.4.0\resources\app.asar\src\display-buffer.js:790:15)
    at TextEditorMarker.module.exports.TextEditorMarker.getHeadScreenPosition (C:\Users\Rasmus\AppData\Local\atom\app-1.4.0\resources\app.asar\src\text-editor-marker.js:179:33)
    at Cursor.module.exports.Cursor.getScreenPosition (C:\Users\Rasmus\AppData\Local\atom\app-1.4.0\resources\app.asar\src\cursor.js:73:26)
    at TextEditor.module.exports.TextEditor.getCursorScreenPosition (C:\Users\Rasmus\AppData\Local\atom\app-1.4.0\resources\app.asar\src\text-editor.js:1432:35)
    at Function.fixString (C:/Users/Rasmus/.atom/packages/linter-jscs/src/linter-jscs.js:172:35)
    at C:/Users/Rasmus/.atom/packages/linter-jscs/src/linter-jscs.js:85:16
    at Function.module.exports.Emitter.simpleDispatch (C:\Users\Rasmus\AppData\Local\atom\app-1.4.0\resources\app.asar\node_modules\event-kit\lib\emitter.js:25:14)
    at Emitter.module.exports.Emitter.emit (C:\Users\Rasmus\AppData\Local\atom\app-1.4.0\resources\app.asar\node_modules\event-kit\lib\emitter.js:125:28)
    at TextBuffer.module.exports.TextBuffer.saveAs (C:\Users\Rasmus\AppData\Local\atom\app-1.4.0\resources\app.asar\node_modules\text-buffer\lib\text-buffer.js:1067:27)
    at TextBuffer.module.exports.TextBuffer.save (C:\Users\Rasmus\AppData\Local\atom\app-1.4.0\resources\app.asar\node_modules\text-buffer\lib\text-buffer.js:1037:19)
    at TextEditor.module.exports.TextEditor.save (C:\Users\Rasmus\AppData\Local\atom\app-1.4.0\resources\app.asar\src\text-editor.js:602:26)
    at Pane.module.exports.Pane.saveItem (C:\Users\Rasmus\AppData\Local\atom\app-1.4.0\resources\app.asar\src\pane.js:534:18)
    at Pane.module.exports.Pane.saveActiveItem (C:\Users\Rasmus\AppData\Local\atom\app-1.4.0\resources\app.asar\src\pane.js:517:19)
    at Workspace.module.exports.Workspace.saveActivePaneItem (C:\Users\Rasmus\AppData\Local\atom\app-1.4.0\resources\app.asar\src\workspace.js:643:35)
    at atom-workspace.commandRegistry.add.core:save (C:\Users\Rasmus\AppData\Local\atom\app-1.4.0\resources\app.asar\src\register-default-commands.js:173:32)
    at CommandRegistry.module.exports.CommandRegistry.handleCommandEvent (C:\Users\Rasmus\AppData\Local\atom\app-1.4.0\resources\app.asar\src\command-registry.js:260:29)
    at C:\Users\Rasmus\AppData\Local\atom\app-1.4.0\resources\app.asar\src\command-registry.js:3:61
    at KeymapManager.module.exports.KeymapManager.dispatchCommandEvent (C:\Users\Rasmus\AppData\Local\atom\app-1.4.0\resources\app.asar\node_modules\atom-keymap\lib\keymap-manager.js:536:16)
    at KeymapManager.module.exports.KeymapManager.handleKeyboardEvent (C:\Users\Rasmus\AppData\Local\atom\app-1.4.0\resources\app.asar\node_modules\atom-keymap\lib\keymap-manager.js:359:22)
    at WindowEventHandler.module.exports.WindowEventHandler.handleDocumentKeydown (C:\Users\Rasmus\AppData\Local\atom\app-1.4.0\resources\app.asar\src\window-event-handler.js:97:36)
    at HTMLDocument.<anonymous> (C:\Users\Rasmus\AppData\Local\atom\app-1.4.0\resources\app.asar\src\window-event-handler.js:3:61)
```

### Commands

```
  5x -0:34.2.0 editor:move-to-end-of-word (atom-text-editor.editor.is-focused)
     -0:33 core:move-right (atom-text-editor.editor.is-focused)
     -0:30.4.0 editor:move-to-end-of-word (atom-text-editor.editor.is-focused)
     -0:28.1.0 core:move-left (atom-text-editor.editor.is-focused)
  2x -0:26.8.0 core:move-right (atom-text-editor.editor.is-focused)
     -0:22.6.0 core:save (atom-text-editor.editor.is-focused)
     -0:18.6.0 core:move-down (atom-text-editor.editor.is-focused)
     -0:15.4.0 core:backspace (atom-text-editor.editor.is-focused)
     -0:12.4.0 pane:split-right (atom-text-editor.editor.is-focused)
     -0:05.5.0 core:move-left (atom-text-editor.editor.is-focused)
     -0:05.3.0 autocomplete-plus:activate (atom-text-editor.editor.is-focused)
     -0:04.7.0 autocomplete-plus:confirm (atom-text-editor.editor.is-focused.autocomplete-active)
     -0:04.5.0 refactor:done (atom-text-editor.editor.is-focused)
     -0:04.5.0 editor:newline (atom-text-editor.editor.is-focused)
     -0:03.7.0 core:backspace (atom-text-editor.editor.is-focused)
  3x -0:03.1.0 core:save (atom-text-editor.editor.is-focused)
```

### Config

```json
{
  "core": {
    "autoHideMenuBar": true,
    "disabledPackages": [
      "terminal-plus",
      "atom-typescript",
      "angularjs"
    ],
    "followSymlinks": false,
    "themes": [
      "atom-material-ui",
      "code-dark-syntax"
    ]
  },
  "linter-jscs": {
    "configPath": ".jscs.yaml",
    "displayAs": "warning",
    "esnext": true,
    "fixOnSave": true
  }
}
```

### Installed Packages

```coffee
# User
atom-jade, v0.3.0
atom-material-ui, v0.8.0
atom-ternjs, v0.12.2
code-dark-syntax, v0.4.0
editorconfig, v1.2.4
fonts, v1.0.0
js-refactor, v0.6.0
linter, v1.11.3
linter-jscs, v3.4.0
minimap, v4.19.0
pane-split-moves-tab, v0.0.3
refactor, v0.6.0

# Dev
No dev packages
```

**Comment by axit-rasmus** [comment]
Steps to reproduce:
1. use plane split (any direction)
2. save

**Comment by casesandberg** [comment]
Just tried reproducing this in a dozen different ways and cant seem to.

**Comment by gleslie2008** [comment]
This has been occurring to me every couple minutes in split-pane view.  Is there more info I can provide?

**Atom Version**: 1.5.4
**System**: Mac OS X 10.11.3
**Thrown From**: [linter-jscs](https://github.com/AtomLinter/linter-jscs) package, v3.4.8


### Stack Trace

Uncaught Error: This TextEditor has been destroyed

```
At /Applications/Atom.app/Contents/Resources/app.asar/src/pane.js:825

Error: This TextEditor has been destroyed
    at DisplayBuffer.module.exports.DisplayBuffer.screenPositionForBufferPosition (/Applications/Atom.app/Contents/Resources/app.asar/src/display-buffer.js:790:15)
    at TextEditorMarker.module.exports.TextEditorMarker.getHeadScreenPosition (/Applications/Atom.app/Contents/Resources/app.asar/src/text-editor-marker.js:179:33)
    at Cursor.module.exports.Cursor.getScreenPosition (/Applications/Atom.app/Contents/Resources/app.asar/src/cursor.js:73:26)
    at TextEditor.module.exports.TextEditor.getCursorScreenPosition (/Applications/Atom.app/Contents/Resources/app.asar/src/text-editor.js:1460:35)
    at Function.fixString (/Users/gleslie/.atom/packages/linter-jscs/src/linter-jscs.js:237:35)
    at /Users/gleslie/.atom/packages/linter-jscs/src/linter-jscs.js:89:18
    at Function.module.exports.Emitter.simpleDispatch (/Applications/Atom.app/Contents/Resources/app.asar/node_modules/event-kit/lib/emitter.js:25:14)
    at Emitter.module.exports.Emitter.emit (/Applications/Atom.app/Contents/Resources/app.asar/node_modules/event-kit/lib/emitter.js:125:28)
    at TextBuffer.module.exports.TextBuffer.saveAs (/Applications/Atom.app/Contents/Resources/app.asar/node_modules/text-buffer/lib/text-buffer.js:1045:20)
    at TextBuffer.module.exports.TextBuffer.save (/Applications/Atom.app/Contents/Resources/app.asar/node_modules/text-buffer/lib/text-buffer.js:1037:19)
    at TextEditor.module.exports.TextEditor.save (/Applications/Atom.app/Contents/Resources/app.asar/src/text-editor.js:630:26)
    at Pane.module.exports.Pane.saveItem (/Applications/Atom.app/Contents/Resources/app.asar/src/pane.js:545:18)
    at Pane.module.exports.Pane.saveActiveItem (/Applications/Atom.app/Contents/Resources/app.asar/src/pane.js:528:19)
    at Workspace.module.exports.Workspace.saveActivePaneItem (/Applications/Atom.app/Contents/Resources/app.asar/src/workspace.js:649:35)
    at atom-workspace.commandRegistry.add.core:save (/Applications/Atom.app/Contents/Resources/app.asar/src/register-default-commands.js:173:32)
    at CommandRegistry.module.exports.CommandRegistry.handleCommandEvent (/Applications/Atom.app/Contents/Resources/app.asar/src/command-registry.js:260:29)
    at /Applications/Atom.app/Contents/Resources/app.asar/src/command-registry.js:3:61
    at KeymapManager.module.exports.KeymapManager.dispatchCommandEvent (/Applications/Atom.app/Contents/Resources/app.asar/node_modules/atom-keymap/lib/keymap-manager.js:570:16)
    at KeymapManager.module.exports.KeymapManager.handleKeyboardEvent (/Applications/Atom.app/Contents/Resources/app.asar/node_modules/atom-keymap/lib/keymap-manager.js:378:22)
    at WindowEventHandler.module.exports.WindowEventHandler.handleDocumentKeydown (/Applications/Atom.app/Contents/Resources/app.asar/src/window-event-handler.js:97:36)
    at HTMLDocument.<anonymous> (/Applications/Atom.app/Contents/Resources/app.asar/src/window-event-handler.js:3:61)
```

### Commands

```
  4x -8:24.6.0 core:move-right (atom-text-editor.editor.is-focused)
  5x -8:23.7.0 core:backspace (atom-text-editor.editor.is-focused)
     -7:47.2.0 core:save (atom-text-editor.editor.is-focused)
  2x -2:47 editor:newline (atom-text-editor.editor.is-focused)
  6x -2:45.3.0 core:backspace (atom-text-editor.editor.is-focused.autocomplete-active)
     -2:41.9.0 editor:newline (atom-text-editor.editor.is-focused)
  4x -2:39.2.0 core:backspace (atom-text-editor.editor.is-focused)
  6x -2:35 core:save (atom-text-editor.editor.is-focused)
     -1:07.9.0 core:copy (atom-text-editor.editor.is-focused)
     -1:07.1.0 editor:newline (atom-text-editor.editor.is-focused)
     -1:07.0 core:paste (atom-text-editor.editor.is-focused)
     -1:05.7.0 editor:newline (atom-text-editor.editor.is-focused)
 14x -0:58.2.0 core:backspace (atom-text-editor.editor.is-focused.autocomplete-active)
     -0:41.6.0 editor:newline (atom-text-editor.editor.is-focused)
 14x -0:37.1.0 core:backspace (atom-text-editor.editor.is-focused.autocomplete-active)
     -0:01.1.0 core:save (atom-text-editor.editor.is-focused)
```

### Config

```json
{
  "core": {},
  "linter-jscs": {
    "fixOnSave": true,
    "onlyConfig": true
  }
}
```

### Installed Packages

```coffee
# User
linter, v1.11.3
linter-jscs, v3.4.8
markdown-preview-plus, v2.2.2

# Dev
No dev packages
```

**Comment by rricamar** [comment]
Hi,
Same error here:
```
System: Ubuntu 14.04
Atom version: 1.7.2
linter: 1.11.4
linterjscs: 3.5.0
```

No special config used,
Also using split panels

```
Error: This TextEditor has been destroyed
    at DisplayBuffer.module.exports.DisplayBuffer.screenPositionForBufferPosition (/usr/share/atom/resources/app.asar/src/display-buffer.js:794:15)
    at TextEditorMarker.module.exports.TextEditorMarker.getHeadScreenPosition (/usr/share/atom/resources/app.asar/src/text-editor-marker.js:179:33)
    at Cursor.module.exports.Cursor.getScreenPosition (/usr/share/atom/resources/app.asar/src/cursor.js:73:26)
    at TextEditor.module.exports.TextEditor.getCursorScreenPosition (/usr/share/atom/resources/app.asar/src/text-editor.js:1490:35)
    at Function.fixString (/home/rricamar/.atom/packages/linter-jscs/src/linter-jscs.js:259:35)
    at /home/rricamar/.atom/packages/linter-jscs/src/linter-jscs.js:95:18
    at Function.module.exports.Emitter.simpleDispatch (/usr/share/atom/resources/app.asar/node_modules/event-kit/lib/emitter.js:25:14)
    at Emitter.module.exports.Emitter.emit (/usr/share/atom/resources/app.asar/node_modules/event-kit/lib/emitter.js:125:28)
    at TextBuffer.module.exports.TextBuffer.saveAs (/usr/share/atom/resources/app.asar/node_modules/text-buffer/lib/text-buffer.js:1082:20)
    at TextBuffer.module.exports.TextBuffer.save (/usr/share/atom/resources/app.asar/node_modules/text-buffer/lib/text-buffer.js:1074:19)
    at TextEditor.module.exports.TextEditor.save (/usr/share/atom/resources/app.asar/src/text-editor.js:660:26)
    at Pane.module.exports.Pane.saveItem (/usr/share/atom/resources/app.asar/src/pane.js:718:18)
    at Pane.module.exports.Pane.saveActiveItem (/usr/share/atom/resources/app.asar/src/pane.js:701:19)
    at Workspace.module.exports.Workspace.saveActivePaneItem (/usr/share/atom/resources/app.asar/src/workspace.js:684:35)
    at atom-workspace.commandRegistry.add.core:save (/usr/share/atom/resources/app.asar/src/register-default-commands.js:214:32)
    at CommandRegistry.module.exports.CommandRegistry.handleCommandEvent (/usr/share/atom/resources/app.asar/src/command-registry.js:260:29)
    at /usr/share/atom/resources/app.asar/src/command-registry.js:3:61
    at KeymapManager.module.exports.KeymapManager.dispatchCommandEvent (/usr/share/atom/resources/app.asar/node_modules/atom-keymap/lib/keymap-manager.js:580:16)
    at KeymapManager.module.exports.KeymapManager.handleKeyboardEvent (/usr/share/atom/resources/app.asar/node_modules/atom-keymap/lib/keymap-manager.js:388:22)
    at WindowEventHandler.module.exports.WindowEventHandler.handleDocumentKeyEvent (/usr/share/atom/resources/app.asar/src/window-event-handler.js:98:36)
    at HTMLDocument.<anonymous> (/usr/share/atom/resources/app.asar/src/window-event-handler.js:3:61)
```

**Comment by axit-rasmus** [comment]
Can this be reopened? It is still an issue.

**Comment by amunro** [comment]
Hey @casesandberg. Was able to nail down reproduction steps for you: 

Atom Version: 1.7.3
System: Mac OS X 10.11.3
Thrown From: linter-jscs package, v4.0.0

Steps for reproduction:

1. Open two files
2. From File 2s tab "Split Right" from the context menu
3. Close original File 2 tab
4. Offend JSCS and try to save

![umfo7rdobt](https://cloud.githubusercontent.com/assets/266702/15203794/13380666-17d5-11e6-9fcd-892753bc004d.gif)

**Comment by Arcanemagus** [comment]
Reopening, I'm assuming you have Fix on Save enabled @amunro?

It's going to be a while before I can look into this, if one of you wants to take a stab at tracking this down the issue has to be with how it is registering the hook [here](https://github.com/AtomLinter/linter-jscs/blob/v4.0.0/src/linter-jscs.js#L85-L86), but I don't know of a better way...

**Comment by Arcanemagus** [comment]
A fix is published in v4.0.2 thanks to @axit-rasmus, please report back if you are still having issues!
